File: Leetcode100/leetcode100_python/UniquePaths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [0 for _ in range(m)]
        for i in range(n):
            for j in range(m):
                if i == 0 or j == 0:
                    dp[j] = 1
                else:
                    dp[j] = dp[j] + dp[j - 1]
        return dp[m - 1]

if __name__ == "__main__":
    m = int(input())
    n = int(input())
    solution = Solution()
    print(solution.uniquePaths(m, n))

File: Leetcode100/leetcode100/src/main/java/com/jiang/leetcode100/unique_paths/UniquePaths.java
package main.java.com.jiang.leetcode100.unique_paths;

import java.util.Scanner;

class Solution {
    public int uniquePaths(int m, int n) {
        int[][] dp = new int[n][m];
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                if(i == 0 || j == 0) dp[i][j] = 1;
                else {
                    dp[i][j] = dp[i - 1][j] + dp[i][j - 1];
                }
            }
        }
        return dp[n - 1][m - 1];
    }
}

public class UniquePaths {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int m = scanner.nextInt();
        int n = scanner.nextInt();
        scanner.close();
        System.out.println(new Solution().uniquePaths(m, n));
    }
}